pkg/ccl/backupccl/backup_processor_planning.go

```diff
@@ -67,13 +67,16 @@ func distBackupPlanSpecs(
 	}
 
 	if encryption != nil && encryption.Mode == jobspb.EncryptionMode_KMS {
-		kms, err := cloud.KMSFromURI(ctx, encryption.KMSInfo.Uri, &backupencryption.BackupKMSEnv{
-			Settings: execCfg.Settings,
-			Conf:     &execCfg.ExternalIODirConfig,
-		})
+		kms, err := cloud.KMSFromURI(ctx, encryption.KMSInfo.Uri, kmsEnv)
 		if err != nil {
 			return nil, err
 		}
+		defer func() {
+			err := kms.Close()
```

Comment:
Will this error for a non-external-connection KMS?

Reply:
This doesn't have to do with this diff, this was wrong before external connections existed. I just saw it manifest as leaked goroutines in my test. We should always close a KMS after we're done using it. I actually pulled this fix out into its own PR - https://github.com/cockroachdb/cockroach/pull/85182